docs/tutorials/azure.md

```diff
@@ -319,6 +321,136 @@ kubectl patch deployment external-dns --namespace "default" --patch \
+```
+
+#### Create a federated identity credential
+
+A binding between the managed identity and the ExternalDNS service account needs to be setup by creating a federated identity resource:
+
+```bash
+$ OIDC_ISSUER_URL="$(az aks show -n myAKSCluster -g myResourceGroup --query "oidcIssuerProfile.issuerUrl" -otsv)"
+
+$ az identity federated-credential create --name ${IDENTITY_NAME} --identity-name ${IDENTITY_NAME} --resource-group $AZURE_AKS_RESOURCE_GROUP} --issuer "$OIDC_ISSUER_URL" --subject "system:serviceaccount:default:external-dns"
+```
+
+NOTE: make sure federated credential refers to correct namespace and service account (`system:serviceaccount:<NAMESPACE>:<SERVICE_ACCOUNT>`)
+
+#### helm
+
+When deploying external-dns with helm, here are the parameters you need to pass:
+
+```yaml
+fullnameOverride: external-dns
+
+serviceAccount:
+  annotations:
+    azure.workload.identity/client-id: <IDENTITY_CLIENT_ID>
+
+podLabels:
+  azure.workload.identity/use: "true"
+
+provider: azure
+
+secretConfiguration:
+  enabled: true
+  mountPath: "/etc/kubernetes/"
+  data:
+    azure.json: |
+      {
+        "subscriptionId": "<SUBSCRIPTION_ID>",
+        "resourceGroup": "<AZURE_DNS_ZONE_RESOURCE_GROUP>",
+        "useWorkloadIdentityExtension": true
+      }
+```
+
+NOTE: make sure the pod is restarted whenever you make a configuration change.
+
+#### kubectl (alternative)
+
+##### Create a configuration file for the managed identity
+
+Create the file `azure.json` with the values from previous steps:
+
+```bash
+cat <<-EOF > /local/path/to/azure.json
+{
+  "subscriptionId": "$(az account show --query id -o tsv)",
+  "resourceGroup": "$AZURE_DNS_ZONE_RESOURCE_GROUP",
+  "useWorkloadIdentityExtension": true
+}
+EOF
+```
+
+Use the `azure.json` file to create a Kubernetes secret:
+
+```bash
+$ kubectl create secret generic azure-config-file --namespace "default" --from-file /local/path/to/azure.json
+```
+
+##### Update labels and annotations on ExternalDNS service account
+
+To instruct Workload Identity webhook to inject a projected token into the ExternalDNS pod, the pod needs to have a label `azure.workload.identity/use: "true"` (before Workload Identity 1.0.0, this label was supposed to be set on the service account instead). Also, the service account needs to have an annotation `azure.workload.identity/client-id: <IDENTITY_CLIENT_ID>`:
+
+To patch the existing serviceaccount and deployment, use the following command:
+
+```bash
+$ kubectl patch serviceaccount external-dns --namespace "default" --patch \
+ "{\"metadata\": {\"annotations\": {\"azure.workload.identity/client-id\": \"${IDENTITY_CLIENT_ID}\"}}}"
+$ kubectl patch deployment external-dns --namespace "default" --patch \
+ '{"spec": {"template": {"metadata": {"labels": {\"azure.workload.identity/use\": \"true\"}}}}}'
+```
+
+NOTE: it's also possible to specify (or override) ClientID through `UserAssignedIdentityID` field in `azure.json`.
```

Comment:
Question: according this line, I kinda understand that "if I do not set the ENV variables right through annotations, I can override the clientId through the value in azure.json"

However, [here](https://github.com/kubernetes-sigs/external-dns/pull/3906/files#diff-e60ec67d16c488548cc9c2a8c5c19c7c0e3b3df803bbc2e4de534308bd5aef1eR105-R110), it feels it is the opposite: if the azure.json does not have the values, we default to the ENV variables.

I tried setting all the values via the azure.json and not using annotations but, the clientId is not fetching properly. May there be a problem in the config somewhere?

Reply:
from the code, it seems the property to set is `ClientID` instead of `UserAssignedIdentityID`:
https://github.com/kubernetes-sigs/external-dns/blob/859892fc72e709d2afe1765f408cabc3b01c9cfa/provider/azure/config.go#L110

Reply:
Yeah, my bad, we should correct either docs or code. Sorry for that.

Reply:
Thanks for the reply, I think this provides quite some insights on why it didn't work. So... which of both options do you prefer? Updating docs or  the code?

I don't mind opening a PR, but I don't want to make a decision without an agreement on what we want.

From my side I wouldn't mind opening a PR to update the docs. Also I can test that `ClientID` and `TenantID` work as expected before reflecting that in the docs.

Reply:
I do encounter the same issue. For me it seems like the default way with annotation of client-id and pod label is not working. I get the error message that the client id is not present. 
To set the clientId in the json file you'll have to set it in the key `aadClientId` and not in `UserAssignedIdentityID` like in the docs. 
However then I get the error message:
`failed to get credentials: failed to create a workload identity token: no token file specified. Check pod configuration or set TokenFilePath in the options`

Reply:
@Sturgelose sorry for the delay in replies, intense period at work and some personal matters.
To be honest, not sure what's the best approach here. On one hand, changing code to `UserAssignedIdentityID` would make more sense in terms of namings. I think I used it in the initial implementation on the old library, but when preparing the new PR, somehow lost the fact that there are two ways to pass client ID in external-dns.
On the other, `aadClientId` is an "easier" change as it doesn't change behaviour.
At the same time, as the documentation is not fully accurate, it's unlikely that many people used overrides. By the way, could you expand on why you need to use explicit override through config instead of k8s annotations?